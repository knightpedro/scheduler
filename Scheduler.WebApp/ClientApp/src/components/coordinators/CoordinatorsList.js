import React from "react";
import { useSelector } from "react-redux";
import { coordinatorsSelectors } from "../../ducks/coordinators";
import { Empty } from "../common";
import { Icon, List } from "semantic-ui-react";

const CoordinatorsList = ({ activeId, filter, handleClick }) => {
  const coordinators = useSelector((state) =>
    coordinatorsSelectors.selectFiltered(state, filter)
  );

  if (coordinators.length === 0)
    return <Empty message="No coordinators found" />;

  return (
    <List selection>
      {coordinators.map((c) => (
        <List.Item
          key={c.id}
          onClick={() => handleClick(c.id)}
          active={c.id === activeId}
        >
          <Icon name="user" color={c.isActive ? "green" : "red"} />
          <List.Content>
            <List.Header>{c.name}</List.Header>
          </List.Content>
        </List.Item>
      ))}
    </List>
  );
};

export default CoordinatorsList;
