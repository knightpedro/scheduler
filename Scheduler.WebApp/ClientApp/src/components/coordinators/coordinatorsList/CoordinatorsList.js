import React, { useMemo } from "react";
import { Link, generatePath } from "react-router-dom";
import Routes from "../../../routes";
import {
  BasicTable,
  SearchColumnFilter,
  SelectColumnFilter
} from "../../common/tables";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCheck, faTimes } from "@fortawesome/free-solid-svg-icons";

const status = {
  ACTIVE: "Active",
  INACTIVE: "Inactive"
};

const CoordinatorsList = ({ coordinators }) => {
  const createPath = id => generatePath(Routes.coordinators.DETAIL, { id });

  const columns = useMemo(
    () => [
      {
        Header: "Name",
        accessor: "name",
        Cell: ({ row }) => (
          <Link to={createPath(row.original.id)}>{row.values.name}</Link>
        ),
        Filter: SearchColumnFilter
      },
      {
        Header: "Active",
        accessor: "status",
        className: "text-center",
        Cell: ({ cell: { value } }) => (
          <FontAwesomeIcon
            icon={value === status.ACTIVE ? faCheck : faTimes}
            color={value === status.ACTIVE ? "green" : "red"}
          />
        ),
        Filter: SelectColumnFilter,
        filter: "exactText"
      }
    ],
    []
  );

  const data = useMemo(
    () =>
      coordinators.map(coordinator => ({
        id: coordinator.id,
        name: coordinator.name,
        status: coordinator.isActive ? status.ACTIVE : status.INACTIVE
      })),
    [coordinators]
  );

  const initialTableState = useMemo(
    () => ({
      sortBy: [
        {
          id: "name"
        }
      ]
    }),
    []
  );

  return (
    <BasicTable
      columns={columns}
      data={data}
      initialState={initialTableState}
    />
  );
};

export default CoordinatorsList;
