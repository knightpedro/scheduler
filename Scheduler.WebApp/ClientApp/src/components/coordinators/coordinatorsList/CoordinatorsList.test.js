import React from "react";
import { shallow } from "enzyme";
import CoordinatorsList from "./CoordinatorsList";

const coordinatorsStub = [];

describe("CoordinatorsList", () => {
    let wrapper;

    beforeEach(
        () =>
            (wrapper = shallow(
                <CoordinatorsList coordinators={coordinatorsStub} />
            ))
    );

    it("renders correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
