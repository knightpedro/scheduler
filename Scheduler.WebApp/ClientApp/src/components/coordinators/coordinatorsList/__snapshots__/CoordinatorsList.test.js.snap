// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoordinatorsList renders correctly 1`] = `
<BasicTable
  columns={
    Array [
      Object {
        "Cell": [Function],
        "Filter": [Function],
        "Header": "Name",
        "accessor": "name",
      },
      Object {
        "Cell": [Function],
        "Filter": [Function],
        "Header": "Active",
        "accessor": "status",
        "className": "text-center",
        "filter": "exactText",
      },
    ]
  }
  data={Array []}
  initialState={
    Object {
      "sortBy": Array [
        Object {
          "id": "name",
        },
      ],
    }
  }
/>
`;
